src/Magnum/DartIntegration/ConvertShapeNode.cpp

```diff
@@ -26,167 +26,293 @@
 
 #include "ConvertShapeNode.h"
 
+#include <dart/dynamics/BodyNode.hpp>
 #include <dart/dynamics/BoxShape.hpp>
 #include <dart/dynamics/CapsuleShape.hpp>
 #include <dart/dynamics/CylinderShape.hpp>
 #include <dart/dynamics/EllipsoidShape.hpp>
 #include <dart/dynamics/MeshShape.hpp>
 #include <dart/dynamics/ShapeNode.hpp>
+#include <dart/dynamics/SoftBodyNode.hpp>
+#include <dart/dynamics/SoftMeshShape.hpp>
 #include <dart/dynamics/SphereShape.hpp>
-#include <Corrade/PluginManager/Manager.h>
+
 #include <Corrade/Utility/Directory.h>
-#include <Magnum/PixelFormat.h>
-#include <Magnum/TextureFormat.h>
-#include <Magnum/MeshTools/Transform.h>
+
+#include <Magnum/Mesh.h>
+#include <Magnum/MeshTools/CombineIndexedArrays.h>
+#include <Magnum/MeshTools/GenerateFlatNormals.h>
 #include <Magnum/Primitives/Capsule.h>
 #include <Magnum/Primitives/Cube.h>
 #include <Magnum/Primitives/Cylinder.h>
 #include <Magnum/Primitives/Icosphere.h>
 #include <Magnum/Trade/AbstractImporter.h>
-
-#include "Magnum/DartIntegration/DartObject.h"
+#include <Magnum/Trade/ImageData.h>
+#include <Magnum/Trade/MeshData3D.h>
+#include <Magnum/Trade/MeshObjectData3D.h>
+#include <Magnum/Trade/ObjectData3D.h>
+#include <Magnum/Trade/PhongMaterialData.h>
+#include <Magnum/Trade/TextureData.h>
 
 namespace Magnum { namespace DartIntegration {
 
-Containers::Optional<ShapeData> convertShapeNode(dart::dynamics::ShapeNode& shapeNode) {
+Containers::Optional<ShapeData> convertShapeNode(dart::dynamics::ShapeNode& shapeNode, ShapeLoadTypes loadType, Trade::AbstractImporter* importer) {
```

Comment:
The `importer` could be just a reference, not a pointer. Similarly in all functions that call `convertShapeNode()` -- `Object::update()` etc.